tensorflow/python/compiler/tensorrt/trt_convert.py

```diff
@@ -295,25 +325,12 @@ def _get_tensorrt_rewriter_config(conversion_params,
     optimizer.parameter_map["max_batch_size"].i = max_batch_size
   optimizer.parameter_map["use_implicit_batch"].b = use_implicit_batch
 
-  # Disabling optimizers should happen after CopyFrom the template
+  # Disabling optimizers should happen after defining the TF-TRT grappler pass
   # otherwise the template can overwrite the disablement.
   if disable_non_trt_optimizers:
-    off = rewriter_config_pb2.RewriterConfig.OFF
-    rewriter_config_with_trt.layout_optimizer = off
-    rewriter_config_with_trt.constant_folding = off
-    rewriter_config_with_trt.shape_optimization = off
-    rewriter_config_with_trt.remapping = off
-    rewriter_config_with_trt.arithmetic_optimization = off
-    rewriter_config_with_trt.dependency_optimization = off
-    rewriter_config_with_trt.loop_optimization = off
-    rewriter_config_with_trt.function_optimization = off
-    rewriter_config_with_trt.debug_stripper = off
-    rewriter_config_with_trt.disable_model_pruning = True
-    rewriter_config_with_trt.scoped_allocator_optimization = off
-    rewriter_config_with_trt.memory_optimization = (
-        rewriter_config_pb2.RewriterConfig.NO_MEM_OPT)
-    rewriter_config_with_trt.pin_to_host_optimization = off
-    rewriter_config_with_trt.auto_parallel.enable = False
+    rewriter_config_with_trt.MergeFrom(
```

Comment:
The intension to outline this block of code and shorten the current routine is good. But instead of creating another rewriter config and merge it with the current one, I think I would prefer (1) to just outline this block of code that clear the fields to a subroutine (2) or keep the original code for the following reasons:
   For singular fields, MergeFrom will use the values in the MergeFrom parameter to overwrite the values in rewriter_config_with_trt. As such, you will need to make sure that _get_disabled_non_trt_optimizers_rewriter_config will generate a rewrite config with the same values as rewriter_config_with_trt for the fields that you don't want to overwrite. This itself is non-trivial and error-prone. Have you noticed that the added routine _get_disabled_non_trt_optimizers_rewriter_config needs to take care of field that the original block "don't care about"?
  For non-singular fields, MergeFrom with concatenate the values in the two objects. for this reason, you will need to make sure get_disabled_non_trt_optimizers_rewriter_config will generate a rewrite config with  "none" for non-singular fields which is also an error-prone job.

Reply:
I can't apply what you propose " But instead of creating another rewriter config and merge it with the current one, I think I would prefer (1) to just outline this block of code that clear the fields to a subroutine (2)" for this unique reason:  **avoiding code duplication**

I need to be able to "deactivate non-TRT optimizers" in the Offline mode, which requires me to directly to set a Rewriter Config "by hand" with all the optimizations deactivated and all of this in a different file: https://github.com/tensorflow/tensorflow/pull/45620/files#diff-39ad5be340981b91107e595a47a42caf7da5a8436448d64a8914c4696384d5a2R343-R344

Not only this, but you could argue that some customers may use an OfflineMode-like and do not use the TRTGraphConverterV1/V2 and therefore would benefit from an automated mechanism as follows.

**How about the following:**
1. The function takes a rewriter_config as argument
2. Directly modifies what it needs like we did before
3. Return the modified rewriter config

I think this solution allows an "in between" that complies with all requirements. I'm going to push that because it's better than my original proposal, I'm waiting for your final feedback.

Reply:
I think the fact that you need the rewrite_config for the offline mode is related to this bug: https://github.com/tensorflow/tensorflow/blob/1ce28a3f8a7670eda543176c7fe3a78b5db11a1e/tensorflow/python/compiler/tensorrt/test/tf_trt_integration_test_base.py#L439-L443

Ideally, line 441 should be "if graph_state == GraphState.INFERENCE and run_params.convert_online:" and we should use None config for the offline mode. But such a line triggers a bug as state in the TODO. So, may be you want to change the line to:

if graph_state == GraphState.INFERENCE and (run_params.convert_online or offline-with-non-trt-opt-disable):

Will this avoid the need of calling _disable_non_trt_optimizers_in_rewriter_config in the test?